hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/capacity/CalculationContext.java

```diff
@@ -0,0 +1,70 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.yarn.server.resourcemanager.scheduler.capacity;
+
+import org.apache.hadoop.yarn.server.resourcemanager.scheduler.capacity.QueueCapacityVector.QueueCapacityVectorEntry;
+
+/**
+ * A storage class that wraps arguments used in a resource calculation iteration.
+ */
+public class CalculationContext {
+  private final String resourceName;
+  private final QueueCapacityVector.ResourceUnitCapacityType capacityType;
+  private final CSQueue childQueue;
```

Comment:
This might be nitpicking, but it was a bit confusing me. Why is it called child queue? I understand that we create a CalculationContext object for every child queue, however from the CalculationContext's point of view, it's not a child, but the actual queue we are doing the calculation. Calling this a child queue suggests that there is a queue we are working with and it is it's children, however this class only works with the child queue, and there is no "actual queue". So I think it would be better to call it simply queue. To my mind this is the desired terminology:

ParentQueue (parent of the queue we are working with)
    |
Queue (the queue we are working with)
    |
ChildQueue (one of our work queue's children)

Reply:
In my opinion this could lead to other problems. We call the handler with a queue, and then perform calculations on its children. In the ResourceCalculationDriver we name the parent queue as queue, but storing its children as queues in the CalculationContext could be misleading as well.